teos/src/responder.rs

```diff
@@ -336,179 +309,60 @@ impl Responder {
+        let dbm = self.dbm.lock().unwrap();
         let tx_index = self.tx_index.lock().unwrap();
+        let mut carrier = self.carrier.lock().unwrap();
+        let mut rejected = Vec::new();
 
-        for (uuid, (penalty_tx, dispute_tx)) in txs.into_iter() {
-            let status = if let Some(dispute_tx) = dispute_tx {
-                // The tracker was reorged out, and the dispute may potentially not be in the chain (or mempool) anymore.
-                if tx_index.contains_key(&dispute_tx.txid())
-                    | carrier.in_mempool(&dispute_tx.txid())
-                {
-                    // Dispute tx is on chain (or mempool), so we only need to care about the penalty
-                    carrier.send_transaction(&penalty_tx)
-                } else {
-                    // Dispute tx has also been reorged out, meaning that both transactions need to be broadcast.
-                    // DISCUSS: For lightning transactions, if the dispute has been reorged the penalty cannot make it to the network.
-                    // If we keep this general, the dispute can simply be a trigger and the penalty doesn't necessarily have to spend from it.
-                    // We'll keel it lightning specific, at least for now.
-                    let status = carrier.send_transaction(&dispute_tx);
-                    if let ConfirmationStatus::Rejected(e) = status {
-                        log::error!(
-                        "Reorged dispute transaction rejected during rebroadcast: {} (reason: {e})",
-                        dispute_tx.txid()
-                    );
-                        status
+        // TODO: Edit this db call to return only the in-mempool trackers so not to pull useless data.
+        for (uuid, tracker_summary) in dbm.load_tracker_summaries() {
+            if let ConfirmationStatus::InMempoolSince(h) = tracker_summary.status {
+                // Rebroadcast the penalty transaction if it has missed too many confirmations.
+                if (height - h) as u8 >= CONFIRMATIONS_BEFORE_RETRY {
+                    let tracker = dbm.load_tracker(uuid).unwrap();
+                    // If the dispute tx isn't on-chain, this means the breach has been reorged out.
+                    // Rebroadcast the dispute transaction.
+                    if !tx_index.contains_key(&tracker_summary.dispute_txid) {
+                        log::warn!(
+                            "Reorged dispute transaction {} for penalty transaction {} has missed many confirmations",
+                            tracker_summary.dispute_txid, tracker_summary.penalty_txid
+                        );
+                        if let ConfirmationStatus::Rejected(e) =
+                            carrier.send_transaction(&tracker.dispute_tx)
+                        {
+                            // DISCUSS: We are broadcasting the penalty transaction even if the dispute transaction was rejected.
```

Comment:
Yep, I don't think we should. This became pretty intertwined after merging the two methods. I'm OK with them being merged by feels like too much of the logic has changed, especially for something as complex (and maybe brittle) as this